fe/fe-core/src/main/java/com/starrocks/statistic/base/SubFieldColumnStats.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.statistic.base;
+
+import com.starrocks.catalog.Type;
+import com.starrocks.statistic.sample.SampleInfo;
+import org.apache.commons.lang.StringEscapeUtils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SubFieldColumnStats extends PrimitiveTypeColumnStats {
```

Comment:
how to use it ? how to store them in memory if there're thousands of fields in a struct  ?

Reply:
refactor from `com.starrocks.statistic.sample.SubFieldColumnStats`,  will save only one in next pr.  it's a tool-class for generate statistics SQL, don't store it in memory.